src/DotLiquid/StandardFilters.cs

```diff
@@ -839,21 +899,40 @@ public static double Abs(Context context, object input)
         /// <param name="context">The DotLiquid context</param>
         /// <param name="input">Input to be transformed by this filter</param>
         /// <param name="atLeast">Value to apply if more than input</param>
+        /// <returns>The larger of the input values.</returns>
+        [LiquidFilter(MinVersion = SyntaxCompatibility.DotLiquid24)]
         public static object AtLeast(Context context, object input, object atLeast)
         {
-            double n;
-            var inputNumber = Double.TryParse(input.ToString(), NumberStyles.Number, context.CurrentCulture, out n);
+            object val1 = NumericConverter.CoerceToNumericType(input, context.FormatProvider, 0);
+            object val2 = NumericConverter.CoerceToNumericType(atLeast, context.FormatProvider, 0);
+            Type resultType = NumericConverter.GetBinaryResultType(val1.GetType(), val2.GetType());
 
-            double min;
-            var atLeastNumber = Double.TryParse(atLeast.ToString(), NumberStyles.Number, context.CurrentCulture, out min);
-
-            if (inputNumber && atLeastNumber)
+            switch (resultType)
             {
-                return (double)((double)min > (double)n ? min : n);
-            }
-            else
-            {
-                return input;
+                case Type t when t == typeof(decimal):
+                    return Math.Max(Convert.ToDecimal(val1), Convert.ToDecimal(val2));
+                case Type t when t == typeof(double):
+                    return Math.Max(Convert.ToDouble(val1), Convert.ToDouble(val2));
+                case Type t when t == typeof(float):
+                    return Math.Max(Convert.ToSingle(val1), Convert.ToSingle(val2));
+                case Type t when t == typeof(ulong):
+                    return Math.Max(Convert.ToUInt64(val1), Convert.ToUInt64(val2));
+                case Type t when t == typeof(long):
+                    return Math.Max(Convert.ToInt64(val1), Convert.ToInt64(val2));
+                case Type t when t == typeof(uint):

```

Comment:
Besides being very verbose, this introduces a new bug. If I call `1 | at_least 2147483648`, I should get a long. Hence the suggestion to use type promotions to get the greater type of the two. Same goes if I have `1 | at_least 2.5`. Here is a simpler suggestion:

```
	var decMax = Math.Max(decInput, decAtLeast);
	if (decMax.Equals(input))
		return input;
	return atLeast;
```

See DoMathsOperation for case that one is double and can't be converted with OverflowException

Reply:
`1 | atleast 2147483648` does result in `long` - the call to `GetBinaryResultType` does the type promotion (as per `DoMathsOperation`). 

But, `2147483648` is actually a `uint` so it should probably return a `uint`. We can simplify this with using dynamic types.